### PR TITLE
fixed the delete facility button to be hidden and not diabled

### LINKED ISSUE #6608: Hide the Delete Facility for user who don't have access to do it
**Describe the bug**

Currently district admin and above have a delete facility button shown in the manage facility dropdown, even though they don't have permission to delete it

**To Reproduce**

1. Go to any facility
2. Click on manage facility
3. See error

**Expected behavior**

Hide the delete facility button for users, who don't have permission to do it

**Screenshots**

image

### PR DESCRIPTION
**Pull request by xakep8**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 5593dc2</samp>

Restrict `Delete Facility` option to authorized users. Modify `src/Components/Facility/FacilityHome.tsx` to check user type before rendering the option.

## Proposed Changes

- Fixes #6608?
- added `authUser.user_type == "DistrictAdmin" || authUser.user_type == "StateAdmin"` for the check of the delete facility button
- Render the delete facility button only if the authenticated user is a `District Admin` or a `State Admin`

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 5593dc2</samp>

*  Restrict the `Delete Facility` option to users with `DistrictAdmin` or `StateAdmin` roles ([link](https://github.com/coronasafe/care_fe/pull/6677/files?diff=unified&w=0#diff-ad6af8429c47f2623c154aeb1256b603ed30a7b233d485bef095d3ddf7c4b97cL597-R611))

### CODE REVIEW

File: src/Components/Facility/FacilityHome.tsx [hunk 1]
```diff
@@ -594,16 +594,21 @@ export const FacilityHome = (props: any) => {
...
+                    onClick={() => setOpenDeleteDialog(true)}
+                    className="flex items-center gap-3"
+                    icon={<CareIcon className="care-l-trash-alt text-lg" />}
+                    authorizeFor={AuthorizeFor(["DistrictAdmin", "StateAdmin"])}
+                  >
+                    Delete Facility
+                  </DropdownItem>
+                ) : (
+                  <></>
+                )}
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityHome.tsx]: Nice work, but a small change, isn't the authorizeFor redundant now that we are hiding the button in the first place? Authorize for would be a no-op right?

```suggestion
                {authUser.user_type == "DistrictAdmin" ||
                authUser.user_type == "StateAdmin" && (
                  <DropdownItem
                    id="delete-facility"
                    variant="danger"
                    onClick={() => setOpenDeleteDialog(true)}
                    className="flex items-center gap-3"
                    icon={<CareIcon className="care-l-trash-alt text-lg" />}
                  >
                    Delete Facility
                  </DropdownItem>
                )}
```

Comment (xakep8) [reply src/Components/Facility/FacilityHome.tsx]: @rithviknishad I just made the suggested changes could you please take a look

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
image

bug: Even as a Palakkad district admin I am not able to delete a facility

the following things need to be addressed in the PR :
1. Modify the user access as state admin and above can only delete a facility
2. Only empty facilities with no patient data should be allowed to be deleted.

**Comment by xakep8** [comment]
@nihal467 How may I join the EOD call?

**Comment by nihal467** [comment]
@xakep8 make the below changes

- [ ] Modify the user access as state admin and above can only delete a facility
- [ ] Only facilities with no patient data should be allowed to be deleted.
- [ ] In case a user attempts to delete a facility with any type of patient data in it, it shouldn't allow the user to delete it and should throw an error. 

@xakep8 The EOD call links are shared in the Reminders channel of the OHC Community Slack workspace. (" https://join.slack.com/t/rebuildearth/shared_invite/zt-28b6p14th-OLAHD9oR9oFLYeSCitrXYw ") 

_Discussion on EOD call 05/12/2023_

**Pull request closed without merging**